examples/CosaMQTTsecureclient/CosaMQTTsecureclient.ino

```diff
@@ -0,0 +1,225 @@
+ * switching imaginary NEXA switches on and off.
+ * The MQTT client connects to the MQTT broker using username/password.
+ * It also specifies a last will.
+ * 
+ * @section Circuit
+ * This sketch is designed for the Ethernet Shield.
+ * @code
+ *                       W5100/ethernet
+ *                       +------------+
+ * (D10)--------------29-|CSN         |
+ * (D11)--------------28-|MOSI        |
+ * (D12)--------------27-|MISO        |
+ * (D13)--------------30-|SCK         |
+ * (D2)-----[ ]-------56-|IRQ         |
+ *                       +------------+
+ *
+ * @endcode
+ * 
+ * @section Reference
+ * 1. Cosa
+ *    Copyright (c) 2012-2015, Mikael Patel
+ *    https://github.com/mikaelpatel/Cosa
+ *    
+ * 2. Cosa-MQTT
+ *    Copyright (c) 2014-2015, Mikael Patel
+ *    https://github.com/mikaelpatel/Cosa-MQTT
+ *    
+ * This file is part of the Arduino Che Cosa project
+ */
+
+#include <DHCP.h>
+#include <DNS.h>
+#include <MQTT.h>
+#include <W5100.h>
+
+#include "Cosa/Watchdog.hh"
+#include "Cosa/Trace.hh"
+#include "Cosa/IOStream/Driver/UART.hh"
+#include "Cosa/String.hh"
+
+#define DEVMODE
+
+// W5100 Ethernet Controller with MAC-address
+static const uint8_t mac[6] __PROGMEM = { 0xde, 0xad, 0xbe, 0xef, 0xfe, 0xed };
+W5100 ethernet(mac);
+
+// MQTT broker configuration
+#define LOCALBROKER "10.0.1.191"
+#define BROKER LOCALBROKER
+
+// MQTT client configurations
+static const char CLIENT[] __PROGMEM = "Cosa-MQTT secure client";       // Client name
+static const char WILL_TOPIC[] __PROGMEM = "node/gone-offline";         // Last will topic
+static const char WILL_MSG[] __PROGMEM = "Cosa-MQTT secure client";     // Last will message
+static const uint16_t will_qos = 0;                                     // Last will QoS
+static const char USER[] __PROGMEM = "Cosa-MQTT";                       // Username for connection to broker
+static const char PASSWD[] __PROGMEM = "mypwd";                         // Password for connection to broker
+static const uint8_t flag = MQTT::Client::WILL_FLAG |                   // Connection flags; combine flag for will, username and password
+                            MQTT::Client::USER_NAME_FLAG |
+                            MQTT::Client::PASSWORD_FLAG;
+
+/*
+ * MQTT client; Parses incoming topic/message
+ */
+class MQTTClient : public MQTT::Client {
+public:
+  virtual void on_publish(char* topic, void* buf, size_t count);
+};
+
+void
+MQTTClient::on_publish(char* topic, void* buf, size_t count)
+{
+  String toparse((char*) topic);
+  String partition;
+  String payload((char*) buf);
+  
+  #if defined(DEVMODE)
+  // Print the topic and value to trace stream
+  trace << PSTR("on_publish::count = ") << count << endl;
+  trace << PSTR("  topic = ") << topic << endl;
+  trace << PSTR("  payload = ") << payload << endl;
+  #endif
+  
+  // Parse NEXA Switches
+  String key(toparse.substring(0, 24));
+  if (key.equals("nexa/switch/"))
```

Comment:
Not sure I understand what you are suggesting.

However, I do see that the parsing is incorrect. The ```24```should read ```12```. The same goes for line 108, 128 and 131.